runtime/README.md

```diff
@@ -1030,29 +1443,201 @@ The DPE `GET_CERTIFICATE_CHAIN` command shall return the following certificates:
+
+These commands are not meant to be high-performance as they are accessed via mailbox commands.
+
+These mailbox commands provide SHA, HMAC, HKDF, AES, and RNG services.
+Asymmetric cryptographic services are currently only provided through DPE and the `ECDSA384_SIGNATURE_VERIFY` and `LMS_SIGNATURE_VERIFY` mailbox commands, which do not use the CM storage system.
+
+### References
+
+* [SPDM 1.3.1 (DSP0274)](https://www.dmtf.org/sites/default/files/standards/documents/DSP0274_1.3.1.pdf), dated 2024-07-01.
+* [OCP Attestation v1.1](https://docs.google.com/document/d/1wA0hbJdtCpcQ1NvsVsYr2IeCkwQbgC7e/edit)
+* [RFC 5869 (HKDF)](https://www.rfc-editor.org/rfc/rfc5869.html)
+* [RFC 8446](https://datatracker.ietf.org/doc/html/rfc8446) Section 7.4.2 & IEEE 1363 (TLS ECDH secret derivation)
+
+### Contexts
+
+Several of the methods, such as SHA and AES, support contexts, so that multiple users can have in-inflight requests at the same time.
+
+The contexts contain the internal structures necessary to resume operations to support data that may exceed the size of a single mailbox command.
+
+These contexts are intended to be opaque to the user, and will be encrypted and authenticated if they contain sensitive internal data.
+
+### Data Storage
+
+The CM system has its own dedicated storage in DCCM.
+This storage is protected with authenticated encryption using internal keys.
+
+This storage is exposed to users through opaque 256-bit handles, called cryptographic mailbox IDs (CMIDs). These are mapped internally into DCCM storage blocks.
+
+Keys can be imported using `CM_IMPORT` and deleted via `CM_DELETE`.
+
+The entire contents can be cleared with `CM_CLEAR`.
+
+### Storage Design
+
+Data is stored in 16-byte blocks.
+
+Each block's status is tracked in a bitfield, 1 bit per block, with a value of 1 indicating that block is in use.
+
+For example, if we have 32 KB of storage for CM, this requires 256 bytes of additional storage.
+
+### CMIDs
+
+CMIDs require no additional storage.
+
+Unecrypted CMIDs have the following structure:
+
+| **Name**     | **Bits** | **Description**                             |
+| ------------ | -------- | ------------------------------------------- |
+| padding      | 80       | 0                                           |
+| block offset | 24       | Starting block number                       |
+| size         | 24       | Size (in bytes) of the data stored.         |
+|              |          | Does not have to be a multiple of 16 bytes. |
+
+Hence, data for a single CMID must be stored contiguously.
+
+Padding is kept for additional future features.
+
+The CMIDs are AES-256-GCM encrypted, with a 128-bit ciphertext concatenated with the 128-bit AES-GCM tag.
+
+### Storage Keys
+
+The key for the data referenced by a CMID is unique and derived via HKDF from the CDI<sub>RT</sub> with the info being the concatenation of `"CMID" || block offset || size`.
```

Comment:
Could we generate a random number and put it in a keyvault slot at boot? Then whenever we need an AES key we can ask the KV slot of hmac the random seed + CMID metadata

Assuming we don't need these keys to live across cold reset.

Reply:
Yes, that would be a preferred way of doing this, since durability is not required across cold boot.

Reply:
Using a random number in a keyvault slot was my first though, but that doesn't work for OCP LOCK, which needs persistence.

Reply:
Do we zeroize DCCM on power on? (It's SRAM, so I thought it would be persistent.)

But in any case, if LOCK can re-import the key after power on every time, then we can eliminate the persistence. (Though it begs the question of where is the LOCK key being stored if not Caliptra that it can be re-imported any time.)